internal/controllers/user_devices_controller.go

```diff
@@ -1856,6 +1856,173 @@ func (udc *UserDevicesController) PostMintDevice(c *fiber.Ctx) error {
+// @Param       burnRequest  body controllers.MintRequest true "Signature and Token ID"
+// @Success     200
+// @Security    BearerAuth
+// @Router      /user/devices/{userDeviceID}/commands/burn [post]
+func (udc *UserDevicesController) PostBurnDevice(c *fiber.Ctx) error {
+	userDeviceID := c.Params("userDeviceID")
+	userID := helpers.GetUserID(c)
+	logger := helpers.GetLogger(c, udc.log)
+
+	userDevice, err := models.UserDevices(
+		models.UserDeviceWhere.ID.EQ(userDeviceID),
+		qm.Load(qm.Rels(models.UserDeviceRels.VehicleNFT, models.VehicleNFTRels.MintRequest)),
+		qm.Load(qm.Rels(models.UserDeviceRels.UserDeviceAPIIntegrations)),
+	).One(c.Context(), udc.DBS().Reader.DB)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return fiber.NewError(fiber.StatusNotFound, "No device with that ID found.")
+		}
+		return err
+	}
+
+	if userDevice.R.VehicleNFT == nil || userDevice.R.VehicleNFT.TokenID.IsZero() {
+		return fiber.NewError(fiber.StatusBadRequest, "Vehicle not minted.")
+	}
+
+	tknID, ok := userDevice.R.VehicleNFT.TokenID.Int64()
+	if !ok {
+		return fiber.NewError(fiber.StatusConflict, "Unable to parse vehicle token id")
+	}
+
+	user, err := udc.usersClient.GetUser(c.Context(), &pb.GetUserRequest{Id: userID})
+	if err != nil {
+		return err
+	}
+
+	if user.EthereumAddress == nil {
+		return fiber.NewError(fiber.StatusBadRequest, "User does not have an Ethereum address on file.")
+	}
+
+	client := registry.Client{
+		Producer:     udc.producer,
+		RequestTopic: "topic.transaction.request.send",
+		Contract: registry.Contract{
+			ChainID: big.NewInt(udc.Settings.DIMORegistryChainID),
+			Address: common.HexToAddress(udc.Settings.DIMORegistryAddr),
+			Name:    "DIMO",
+			Version: "1",
+		},
+	}
+
+	bvs := registry.BurnVehicleSign{
+		TokenID: big.NewInt(tknID),
+	}
+
+	br := new(BurnRequest)
+	if err := c.BodyParser(br); err != nil {
+		return fiber.NewError(fiber.StatusBadRequest, "Couldn't parse request body.")
+	}
+
+	logger.Info().
+		Interface("httpRequestBody", br).
+		Interface("client", client).
+		Interface("burnVehicleSign", bvs).
+		Interface("typedData", client.GetPayload(&bvs)).
+		Msg("Got request.")
+
+	hash, err := client.Hash(&bvs)
+	if err != nil {
+		return opaqueInternalError
+	}
+
+	sigBytes := common.FromHex(br.Signature)
+	recAddr, err := helpers.Ecrecover(hash, sigBytes)
+	if err != nil {
+		return err
+	}
+
+	realAddr := common.HexToAddress(*user.EthereumAddress)
+	if recAddr != realAddr {
+		return fiber.NewError(fiber.StatusBadRequest, "signature incorrect")
+	}
+
+	requestID := ksuid.New().String()
+	mtr := models.MetaTransactionRequest{
+		ID:     requestID,
+		Status: "Unsubmitted",
+	}
+	err = mtr.Insert(c.Context(), udc.DBS().Writer, boil.Infer())
+	if err != nil {
+		return err
+	}
+
+	userDevice.R.VehicleNFT.BurnRequestID = null.StringFrom(requestID)
+
+	if _, err := userDevice.R.VehicleNFT.Update(c.Context(), udc.DBS().Writer, boil.Infer()); err != nil {
```

Comment:
Probably want to be careful with a transaction here. You do not want someone to double-submit this. See the claim handler for an example.